constructorio-client/src/main/java/io/constructor/client/ConstructorException.java

```diff
@@ -1,11 +1,28 @@
 package io.constructor.client;
 
 public class ConstructorException extends Exception {
+	private Integer errorCode;
+
 	public ConstructorException(String msg) {
 		super(msg);
 	}
 
 	public ConstructorException(Exception e) {
 		super(e);
+		if (e instanceof ConstructorException) {
+			Integer errorCode = ((ConstructorException) e).getErrorCode();
+			if (errorCode != null){
+				this.errorCode = errorCode;
+			}
```

Comment:
This could also be simplified ...

Reply:
```java
Integer errorCode = ((ConstructorException) e).getErrorCode();
if (errorCode != null){
   this.errorCode = errorCode;
}
```
is the same as

```java
this.errorCode = ((ConstructorException) e).getErrorCode();
```

Reply:
As you can tell I am scared off nulls in Java